reflect.go

```diff
@@ -232,12 +243,6 @@ func (r *Reflector) reflectTypeToSchema(definitions Definitions, t reflect.Type)
 		}}
 	}
 
-	if r.TypeMapper != nil {
```

Comment:
I also moved this above. To me it feels like `TypeMapper` is the highest priority override. Then `JSONSchemaType`. And then built-in stuff.